security/pkg/k8s/configmap/configmap.go

```diff
@@ -119,3 +215,91 @@ func (c *Controller) GetCATLSRootCert() (string, error) {
 
 	return rootCert, nil
 }
+
+func (c *Controller) Run(stopCh chan struct{}) {
+	if c.extraTrustAnchorsController != nil {
+		c.extraTrustAnchorsController.Run(stopCh)
+	}
+}
+
+func (c *Controller) WaitForSync(stopCh chan struct{}) {
+	if c.extraTrustAnchorsController != nil {
+		cache.WaitForCacheSync(stopCh, c.extraTrustAnchorsController.HasSynced)
+	}
+}
+
+func (c *Controller) RefreshTrustAnchorsBundle() []byte {
+	c.extraTrustAnchorMu.Lock()
+	defer c.extraTrustAnchorMu.Unlock()
+	return c.refreshTrustAnchorsBundle()
+}
+
+// The caller is responsible for holding the sc.extraTrustAnchorMu lock while calling this method.
+func (c *Controller) refreshTrustAnchorsBundle() []byte {
+	trustAnchorsSet := map[string]struct{}{}
+
+	c.primaryTrustAnchor = c.ca.GetCAKeyCertBundle().GetRootCertPem()
+	primary := strings.TrimSpace(string(c.primaryTrustAnchor))
+	trustAnchorsSet[primary] = struct{}{}
+
+	for _, configMap := range c.extraTrustAnchors {
+		for _, value := range configMap {
+			cert := strings.TrimSpace(value)
+			trustAnchorsSet[cert] = struct{}{}
+		}
+	}
+
+	trustAnchors := make([]string, 0, len(trustAnchorsSet))
+	for trustAnchor := range trustAnchorsSet {
+		trustAnchors = append(trustAnchors, trustAnchor)
+	}
+	sort.Strings(trustAnchors)
+
+	c.combinedTrustAnchorsBundle = []byte(strings.Join(trustAnchors, "\n"))
+
+	return c.combinedTrustAnchorsBundle
```

Comment:
If the CAKCB is storing all the roots, why don't we update the CAKCB in this function?